src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Sortable.java

```diff
@@ -1,10 +1,27 @@
 package io.stargate.sgv2.jsonapi.api.model.command;
 
+import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.SortSpec;
 import io.stargate.sgv2.jsonapi.api.model.command.clause.sort.SortClause;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import java.io.IOException;
 
 /*
  * All the commands that needs SortClause will have to implement this.
  */
 public interface Sortable {
-  SortClause sortClause();
+  /** Accessor for the filter specification in its intermediate for */
+  SortSpec sortSpec();
+
+  default SortClause sortClause(CommandContext<?> ctx) {
+    return sortClause(ctx.schemaObject());
+  }
+
+  default SortClause sortClause(SchemaObject schema) {
```

Comment:
Ideally we would take just `CommandContext`, but there are callers who only have `SchemaObject` (validator) so that'll have to do for `SortClause` handling.